### PR TITLE
Feat/lesq 253/old quiz new comp

### PR DESCRIPTION
**Pull request by kimon-satan**
Transform the old data to match the new data schema.

Check it renders correctly for all question types

## To Test:

b) images in mcq answers

- https://deploy-preview-1841--oak-web-application.netlify.thenational.academy/beta/teachers/programmes/maths-secondary-ks3/units/numbers-and-numerals-6c2e/lessons/mayan-numbers-c4w66d
- [[https://deploy-preview-1841--oak-web-application.netlify.thenational.academy/beta/teachers/programmes/maths-primary-ks1/units/measure[…]to-measure-lengths-using-non-standard-units-part-2-6ngkac](https://deploy-preview-1841--oak-web-application.netlify.thenational.academy/beta/teachers/programmes/maths-primary-ks1/units/measure%5B%E2%80%A6%5Dto-measure-lengths-using-non-standard-units-part-2-6ngkac)](https://deploy-preview-1841--oak-web-application.netlify.thenational.academy/beta/teachers/programmes/maths-primary-ks1/units/measures-1-length-and-mass-0f65/lessons/to-measure-lengths-using-non-standard-units-part-2-6ngkac)
- **exit** [[https://deploy-preview-1841--oak-web-application.netlify.thenational.academy/beta/teachers/programmes/science-secondary-ks3/units/ato[…]-periodic-table-9q9t0p6/lessons/chemical-reactions-p26zu7](https://deploy-preview-1841--oak-web-application.netlify.thenational.academy/beta/teachers/programmes/science-secondary-ks3/units/ato%5B%E2%80%A6%5D-periodic-table-9q9t0p6/lessons/chemical-reactions-p26zu7)](https://deploy-preview-1841--oak-web-application.netlify.thenational.academy/beta/teachers/programmes/science-secondary-ks3/units/atoms-and-the-periodic-table-9q9t0p6/lessons/chemical-reactions-p26zu7)

c) match questions

- **exit** https://deploy-preview-1841--oak-web-application.netlify.thenational.academy/beta/teachers/programmes/science-secondary-ks3/units/cells-u2b1h4c/lessons/cell-structures-fxjz0y
- **starter** {owa_deployment_url}/beta/teachers/programmes/science-secondary-ks3/units/cells-u2b1h4c/lessons/diffusion-ocvf5s

d) order questions

- **starter** {owa_deployment_url}/beta/teachers/programmes/science-secondary-ks3/units/cells-u2b1h4c/lessons/cell-structures-fxjz0y
- **exit** [[{owa_deployment_url}/beta/teachers/programmes/science-secondary-ks3/units/ato[…]-9q9t0p6/lessons/development-of-the-periodic-table-r21gyc]({owa_deployment_url}/beta/teachers/programmes/science-secondary-ks3/units/ato%5B%E2%80%A6%5D-9q9t0p6/lessons/development-of-the-periodic-table-r21gyc)]({owa_deployment_url}/beta/teachers/programmes/science-secondary-ks3/units/atoms-and-the-periodic-table-9q9t0p6/lessons/development-of-the-periodic-table-r21gyc)

e. short answer questions

- **starter** [[{owa_deployment_url}/beta/teachers/programmes/science-secondary-ks3/units/ato[…]-periodic-table-9q9t0p6/lessons/chemical-reactions-p26zu7]({owa_deployment_url}/beta/teachers/programmes/science-secondary-ks3/units/ato%5B%E2%80%A6%5D-periodic-table-9q9t0p6/lessons/chemical-reactions-p26zu7)]({owa_deployment_url}/beta/teachers/programmes/science-secondary-ks3/units/atoms-and-the-periodic-table-9q9t0p6/lessons/chemical-reactions-p26zu7)
- **starter and exit** {owa_deployment_url}/beta/teachers/programmes/biology-secondary-ks4/units/cells-biology-3p8njnj/lessons/osmosis-eymw2x

f) explanatory texts

- **starter** [[{owa_deployment_url}/beta/teachers/programmes/german-secondary-ks4/units/holi[…]sing-accommodation-and-associated-problems-part-13-cmr6ct]({owa_deployment_url}/beta/teachers/programmes/german-secondary-ks4/units/holi%5B%E2%80%A6%5Dsing-accommodation-and-associated-problems-part-13-cmr6ct)]({owa_deployment_url}/beta/teachers/programmes/german-secondary-ks4/units/holidays-and-travel-9ac5/lessons/discussing-accommodation-and-associated-problems-part-13-cmr6ct)
- **exit** [[{owa_deployment_url}/beta/teachers/programmes/spanish-secondary-ks4/units/fre[…]/lessons/talking-about-what-you-usually-do-part-33-cnk32t]({owa_deployment_url}/beta/teachers/programmes/spanish-secondary-ks4/units/fre%5B%E2%80%A6%5D/lessons/talking-about-what-you-usually-do-part-33-cnk32t)]({owa_deployment_url}/beta/teachers/programmes/spanish-secondary-ks4/units/free-time-f937/lessons/talking-about-what-you-usually-do-part-33-cnk32t)

## Acceptance Criteria:

- [x]  Both quiz types render
- [x]  Question Image
- [x]  MCQ no images
- [x]  MCQ with images
- [x]  Match
- [x]  Order
- [x]  Short
- [x]  Explanatory text
- [x]  checkbox
- [x]  Unit tests

## Screenshots

How it used to look (delete if n/a):
{screenshots}

How it should now look:
{screenshots}

### CODE REVIEW

**Review by jack-skerrett-oak — CHANGES_REQUESTED** [review]
So I think the match questions are the wrong way round for the old content. Here the "choices" are displayed on the right, and the "answers" on the left, we want this the other way round (I think we have it the right way round for new quiz data)

Screenshot 2023-08-30 at 17 00 37

File: package.json [hunk 1]
```diff
@@ -159,6 +159,7 @@
     "@wdio/mocha-framework": "^8.15.9",
     "@wdio/spec-reporter": "^8.15.7",
     "browserstack-local": "^1.5.2",
+    "chromedriver": "^116.0.0",
```
[the remark this answers is not in the document]
Comment (kimon-satan) [reply package.json:162]: yes. sorry it shouldn't have ended up in this one but it doesn't do any harm

File: src/node-lib/curriculum-api/transformQuizzes.ts [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+} from "../curriculum-api-2023/queries/lessonOverview/lessonOverview.schema";
+
+import { LegacyQuizData } from "./legacyQuiz.schema";
+
+const questionTypeMap = (questionType?: string | null) => {
+  switch (questionType) {
+    case "text":
+    case undefined:
+    case null:
+      return "explanatory-text" as keyof AnswersSchema;
```
[the remark this answers is not in the document]
Comment (kimon-satan) [reply src/node-lib/curriculum-api/transformQuizzes.ts:16]: I really tried but couldn't find a way . Happy to explore any ideas.

**Comment by kimon-satan** [comment]
> So I think the match questions are the wrong way round for the old content. Here the "choices" are displayed on the right, and the "answers" on the left, we want this the other way round (I think we have it the right way round for new quiz data)
> 
> Screenshot 2023-08-30 at 17 00 37

Swapped now . Please check

**Review by jack-skerrett-oak — APPROVED** [review]

**Pull request merged**